### PR TITLE
Transport review

### PR DESCRIPTION
**Pull request by Bruno007**

### CODE REVIEW

File: draft-ietf-lsr-isis-fast-flooding.xml [hunk 1]
```diff
@@ -784,7 +757,7 @@
           current LSPTxRate.</t>
 
           <t>The flow control algorithm MUST NOT assume the receive
```
Comment (lesginsberg) [inline draft-ietf-lsr-isis-fast-flooding.xml:759]: Please replace all occurrences in this section  of "flow control" with "congestion control".

File: draft-ietf-lsr-isis-fast-flooding.xml [hunk 2]
```diff
@@ -439,17 +439,24 @@
 			  allows for more frequent PSNPs, giving
 			  faster feedback to the sender. Outside of
 			  the burst case, the usual time-based PSNP
-			  approach comes into effect. The LPP SHOULD
-			  also be less than or equal to 90 as this is
+			  approach comes into effect.</t>
+			  <t> The smaller the LPP, the faster the feedback to the sender 
+			  and possibly the higher the rate if the rate is limited by the
+			   end to end RTT (link RTT + time to acknowledge). But the higher
```
[the remark this answers is not in the document]
Comment (Bruno007) [reply draft-ietf-lsr-isis-fast-flooding.xml:445]: Thanks. Changed accordingly

File: draft-ietf-lsr-isis-fast-flooding.xml [hunk 3]
```diff
@@ -498,8 +505,7 @@
         transmitted. It may also make it more likely that a receiver becomes
         overwhelmed by LSP transmissions.</t>
 
-			<t>Therefore implementations SHOULD prioritize the
-        receipt of Hellos and then SNPs over LSPs. Implementations MAY also prioritize IS-IS packets over other less critical protocols.</t>
+			<t>Therefore implementations SHOULD prioritize IS-IS PDU on the way from the incoming interface to the IS-IS process. The relative priority of packets in decreasing order SHOULD be: Hellos, SNPs, LSPs. Implementations MAY also prioritize IS-IS packets over other protocols which are less critical for the router or network, less sensitive to delay or more bursty (e.g., BGP).</t>
```
[the remark this answers is not in the document]
Comment (Bruno007) [reply draft-ietf-lsr-isis-fast-flooding.xml:508]: Thanks done.

File: draft-ietf-lsr-isis-fast-flooding.xml (hunk 1 above)
Comment (Bruno007) [reply draft-ietf-lsr-isis-fast-flooding.xml:759]: OK. I took the liberty to also apply the chane to the time. Please correct me if I'm wrong.

File: draft-ietf-lsr-isis-fast-flooding.xml [hunk 4]
```diff
@@ -545,14 +551,17 @@
 		    rwin never goes negative.
 				</t>
 
+			<t>The RWIN value is of importance when the RTT is the limiting factor for the throughput. In this case the optimal size is the desired LSP rate divided by the RTT. The RTT being the addition of the link RTT plus the time taken by the receiver to acknowledge the first received LSP in its PSNP. 50 or 100 may be reasonable default numbers. As an example, a RWIN of 100 requires a control plane input buffer of 150 ko per neighbor assuming an IS-IS MTU of 1500 octets and limits the throughput to 10000 LSPs per second and per neighbor for a link RTT of 10 ms. With a same RWIN, the throughput limitation is 2000 LSP per second when the RTT is 50ms. That's the maximum throughput assuming no other limitations such as CPU limitations.</t>
```
Comment (gsol10) [inline draft-ietf-lsr-isis-fast-flooding.xml:554]: With a same RWIN/With the same RWIN ?

File: draft-ietf-lsr-isis-fast-flooding.xml [hunk 5]
```diff
@@ -622,26 +596,26 @@
    ----------------+
 					</artwork>
 				</figure>
-				<t>The algorithm starts with cwin = LPP + 1. In the congestion avoidance phase, cwin increases as LSPs are acked: for every acked LSP, cwin += 1 / cwin. Thus, the sending rate is inversely proportional to the RTT. Since the RTT is low in many IS-IS deployments, the sending rate can reach fast rates in short periods of time.</t>
+				<t>The algorithm starts with cwin = cwin0 = LPP + 1. In the congestion avoidance phase, cwin increases as LSPs are acked: for every acked LSP, cwin += 1 / cwin. When LSPs are exchanged, cwin LSPs will be acknowledged in 1 RTT, meaning cwin(t) = t/RTT + cwin0. Since the RTT is low in many IS-IS deployments, the sending rate can reach fast rates in short periods of time.</t>
 
 				<t>When updating cwin, it must not become higher than the number of LSPs waiting to be sent, otherwise the sending will not be paced by the receiving of acks. Said differently, tx pressure is needed to maintain and increase cwin.</t>
```
Comment (gsol10) [inline draft-ietf-lsr-isis-fast-flooding.xml:601]: Can we also add  `cwin, by definition, should not become bigger than the receive window advertised by the receiver.` ?

**Review by gsol10 — COMMENTED** [review]
Thanks a lot for addressing the comments and driving the discussion. Just a few remarks.

**Comment by lesginsberg** [comment]
Guillaume –

Just some editorial remarks on one change you propose.

OLD: For an IS-IS adjacency, the network between two IS-IS neighbors is relatively limited in scope and consists of a link that is typically over-sized compared to the capability of the IS-IS speakers, but may also include components inside both routers such as a switching fabric, line card CPU, and forwarding plane buffers that may experience congestion.

NEW: This path typically includes the point-to-point link between the router which is usually over-sized compared to the capability of the IS-IS speakers, but potentially some internal elements inside each neighbor such as switching fabric, line card CPU, and forwarding plane buffers that may experience congestion.

The new text doesn’t parse well. You use the phrase “between the router” but never define “between the router and what…”

Would you be OK with:

“This path typically includes a point-to-point link between two IS-IS neighbors which is…”

??

Alternatively, if we do not want to completely ignore LANs here, we could say:

““This path typically includes a link between two IS-IS neighbors which is…”

   Les

**Comment by gsol10** [comment]
@lesginsberg agreed on your first proposal. I meant to write between two routerS but yours is more precise anway

File: draft-ietf-lsr-isis-fast-flooding.xml (hunk 5 above)
Comment (Bruno007) [reply draft-ietf-lsr-isis-fast-flooding.xml:601]: Yes. Feel a priory obvious but that's part of the spec and better safe than sorry.
Changed to "for every acked LSP, cwin += 1 / cwin without exceeding RWIN.". Please feel free to comment/correct.

**Comment by Bruno007** [comment]
> @lesginsberg agreed on your first proposal. I meant to write between two routerS but yours is more precise anway
+1 thanks

**Comment by Bruno007** [comment]
Thank you Guillaume for your review and comments. Especially as you were busy this week.
I've just pushed a commit to include your comments plus the one from Les.

I've also pushed another commit with the below change:
OLD: The LSP Transmission Interval is an advertisement of the receiver's steady-state LSP reception rate.
NEW: The LSP Transmission Interval is an advertisement of the receiver's sustainable LSP reception rate. This rate may be safely used by a sender which do not support the flow control or congestion algorithm. It may also be used as the minimal safe rate by flow control or congestion algorithms in unexpected cases, e.g., when the receiver is not acknowledging LSPs anymore.
In https://datatracker.ietf.org/doc/html/draft-ietf-lsr-isis-fast-flooding-06#name-lsp-transmission-interval-s

(note: this is expected to match the existing Cisco  isis lsp-interval  CLI)

2 reasons:

  *   If the receiver is not acknowledging LSP anymore (possibly because it received none as al were dropped) the sender needs to be able to send LSP to sync the LSDB and also trigger new PSNP. This is typically covered in TCP but I could not find the equivalent in our congestion/flow control sections.
  *   "steady-stade" looks like the max rate (receiver is not falling behind but also not draining its queue) while we mean the safe/min value.

Thanks
Regards,
--Bruno

**Pull request merged**